lib/protobuf/message/serialization.rb

```diff
@@ -48,10 +48,12 @@ def decode_from(stream)
       # Encode this message
       #
       def encode
-        stream = ::StringIO.new
-        stream.set_encoding(::Protobuf::Field::BytesField::BYTES_ENCODING)
-        encode_to(stream)
-        stream.string
+        @memoized_encoded ||= begin
```

Comment:
if the value is going to be memoized then we should call the instance var the same as the method